plutus-core/testlib/Test/Tasty/Extras.hs

```diff
@@ -50,31 +65,116 @@ ghcVersion = showVersion compilerVersion
 makeVersionedFilePath :: [FilePath] -> FilePath -> FilePath
 makeVersionedFilePath path file = joinPath path </> ghcVersion </> file
 
--- | A 'TestTree' of tests under some name prefix.
-type TestNested = Reader [FilePath] TestTree
+{- | A monad allowing one to emit elements of type @a@. Semantically equivalent to
+@Writer (DList a) r@, but:
+
+1. is faster, being based on the Church-encoded free monad
+2. implements 'Monoid', so that all the "Data.Foldable" convenience is supported
+3. has better ergonomics as it doesn't require the user to wrap @a@ values into 'DList's
+
+This type is also semantically equivalent to @Stream (Of a) Identity r@.
+
+Useful for monadically creating tree-like structures, for example the following
```

Comment:
I need that for another PR to update a huge `do`-block, so that it stays the same but some of the tests become golden.